libs/react-ui/src/lib/main-navigation/main-navigation.tsx

```diff
@@ -0,0 +1,122 @@
+import { FC, MouseEvent, useEffect, useState } from 'react';
+
+import { NavLink, useLocation } from 'react-router-dom';
+import { MdClose, MdKeyboardArrowDown } from 'react-icons/md';
+import { useTranslation } from 'react-i18next';
+import { useQuery } from '@tanstack/react-query';
+import clsx from 'clsx';
+import { MdOutlineForum, MdOutlineAdb, MdOutlineEqualizer, MdSettings, MdOutlineMonitorWeight } from 'react-icons/md';
+import { Icon } from 'components';
+import './main-navigation.scss';
+
+interface MenuItem {
+  id?: string;
+  label: TranslatedLabel;
+  path: string | null;
+  target?: '_blank' | '_self';
+  children?: MenuItem[];
+}
+
+interface TranslatedLabel {
+  [lang: string] : string;
+}
+
+const MainNavigation: FC<{items: MenuItem[]}> = ({items}) => {
+
+  const { t } = useTranslation();
+  const [menuItems, setMenuItems] = useState<MenuItem[]>([]);
+  const menuIcons = [
+    {
+      id: 'conversations',
+      icon: <MdOutlineForum />,
+    },
+    {
+      id: 'training',
+      icon: <MdOutlineAdb />,
+    },
+    {
+      id: 'analytics',
+      icon: <MdOutlineEqualizer />,
+    },
+    {
+      id: 'settings',
+      icon: <MdSettings />,
+    },
+    {
+      id: 'monitoring',
+      icon: <MdOutlineMonitorWeight />,
+    },
+  ];
+
+  const { data } = useQuery({
+    queryKey: ['cs-get-user-role', 'prod', items[0]?.id],
+    onSuccess: (res: any) => {
+      const filteredItems = items.filter((item) => {
+        const role = res.data.get_user[0].authorities[0]
+        switch (role) {
+          case 'ROLE_ADMINISTRATOR': return item.id
+          case 'ROLE_SERVICE_MANAGER': return item.id != 'settings' && item.id != 'training'
+          case 'ROLE_CUSTOMER_SUPPORT_AGENT': return item.id != 'settings' && item.id != 'analytics'
+          case 'ROLE_CHATBOT_TRAINER': return item.id != 'settings' && item.id != 'conversations'
+          case 'ROLE_ANALYST': return item.id == 'analytics'
+          case 'ROLE_UNAUTHENTICATED': return
```

Comment:
If user level checking is done in frontend, doesn't this make possible for malicious user (even unauthicated one, maybe ?) to access modules otherwise denied to them by changing variables runtime ?

I'm not exactly familiar with this project's security requirements so if this question is irrelevant, let me know.

Reply:
@RayDNoper GUI is just a facade. No matter what it allows, Ruuter is always the middleman to make appropriate validations on the back-end side.